Add some GHCup codes (#371)
haskellfoundation/error-message-index: message-index/messages/GHCup-00841/index.md

---
title: A process exited prematurely

> summary: GHCup executed a subprocess, which did not execute successfully

Reviewer: ```suggestion
summary: GHCup executed a subprocess that did not complete successfully
```

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  145M  100  145M    0     0  42.0M      0  0:00:03  0:00:03 --:--:-- 42.0M
[ Info  ] verifying digest of: ghc-8.4.4-x86_64-fedora27-linux.tar.xz
[ Info  ] Unpacking: ghc-8.4.4-x86_64-fedora27-linux.tar.xz to /home/hasufell/.ghcup/tmp/ghcup-3ee9976c7bb53f16
[ Info  ] Installing GHC (this may take a while)
[ ghc-configure ] checking whether ln -s works... yes
[ ghc-configure ] checking for gsed... sed
[ ghc-configure ] checking whether the C compiler works... no
[ ghc-configure ] configure: error: in `/home/hasufell/.ghcup/tmp/ghcup-3ee9976c7bb53f16/ghc-8.4.4':
[ ghc-configure ] configure: error: C compiler cannot create executables
[ ghc-configure ] See `config.log' for more details
[ Error ] [GHCup-00841] Process "sh" with arguments ["./configure",
[ ...   ]                              "--prefix=/home/hasufell/.ghcup/ghc/8.4.4"] failed with exit code 77.
[ Error ] Also check the logs in /home/hasufell/.ghcup/logs
```

> This commonly happens on darwin too if xcode command line tools are not installed or if llvm is not installed on darwin M1 (for GHC versions that don't have native code generator, such as 8.10.7 or 9.0.2).

Reviewer: ```suggestion
This commonly happens on macOS too if Xcode command line tools are not installed or if LLVM is not installed on Apple Silicon machines (for GHC versions that don't have native code generator, such as 8.10.7 or 9.0.2).
``` 

I think this clarifies what systems are for users who don't yet have the Deep Unix view of things.

> This commonly happens on macOS too if Xcode command line tools are not installed or if LLVM is not installed on Apple Silicon machines (for GHC versions that don't have native code generator, such as 8.10.7 or 9.0.2).

Reviewer: ```suggestion
This commonly happens on macOS too if Xcode command line tools are not installed or if LLVM is not installed on Apple Silicon machines (for GHC versions that don't have a native code generator, such as 8.10.7 or 9.0.2).
```

> See the following [SO answer on xcode command line tools](https://stackoverflow.com/a/9329325) and the [brew llvm formula page](https://formulae.brew.sh/formula/llvm) (make sure to read the instructions after installation to adjust PATH).

Reviewer: ```suggestion
See the following [Stack Overflow answer on Xcode command line tools](https://stackoverflow.com/a/9329325) and the [Brew LLVM formula page](https://formulae.brew.sh/formula/llvm) (make sure to read the instructions after installation to adjust the `PATH` environment variable).
```
Capitalization suggestions plus giving a Google hook for those who don't yet know PATH
